### ISSUE TITLE
Creating new IOBuffer is not necessary

### ISSUE THREAD

**Issue by jlapeyre** [davidanthoff/StringBuilders.jl#3]
Creating a new IOBuffer is not necessary here
https://github.com/davidanthoff/StringBuilders.jl/blob/47380e71e4003e4c812e6d196d927d191bfba059/src/StringBuilders.jl#L23
because the IOBuffer is reset to its initial state after `take!`.

On the other hand, there is probably no significant penalty for doing it.

**Comment by davidanthoff** [comment]
Oh, so ``IOBuffer`` automatically starts with a new memory location once one has called ``take!``?

**Comment by jlapeyre** [comment]
Yes. You can read it in `? take!`.  I tried deleting the line and your tests still pass.

I agree that something like `StringBuilder` should be officially sanctioned in some way, so we can rely on it. Just a couple of weeks ago, I did not use IOBuffer because I was in a hurry and couldn't bother to look it up again.

**Comment by davidanthoff** [comment]
Cool. would be great if you could open a PR for this!
